code/game/machinery/dance_machine.dm

```diff
@@ -460,17 +463,18 @@
 		var/sound/song_played = sound(selection.song_path)
 
 		for(var/mob/M in range(10,src))
-			if(!(M in rangers))
-				rangers[M] = TRUE
-				M.playsound_local(get_turf(M), null, 100, channel = CHANNEL_JUKEBOX, S = song_played)
-			if(prob(5+(allowed(M) * 4)) && M.canmove && isliving(M))
```

Comment:
The allowed checks for access, always returns 1 as the machine doesn't require access, so just stortened that to 9.